kevm-pyk/src/kevm_pyk/kproj/evm-semantics/abi.md

```diff
@@ -510,21 +513,34 @@ where `F1 : F2 : F3 : F4` is the (two's complement) byte-array representation of
 
     rule #isStaticType(#array(_, _, _)) => false
 
+    rule #isStaticType(#tuple( ARGS )) => notBool #hasDynamicType(ARGS)
+
+    syntax Bool ::= #hasDynamicType(TypedArgs) [function, total]
+ // ------------------------------------------------------------
+    rule #hasDynamicType(.TypedArgs) => false
+    rule #hasDynamicType(T, TS) => #hasDynamicType(TS) requires #isStaticType(T)
+    rule #hasDynamicType(T,  _) => true requires notBool #isStaticType(T)
+
     syntax Int ::= #sizeOfDynamicType ( TypedArg ) [function]
```

Comment:
Is this function total?

Reply:
No, because it has rules only for `#bytes` , `#tuple` and `#array`.